### PR TITLE
Add get_account_congestion route

### PR DESCRIPTION
**Pull request by thibault-martinez**
Closes https://github.com/iotaledger/iota-sdk/issues/1050

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/types/api/core/response.rs [hunk 1]
```diff
@@ -343,3 +343,22 @@ pub struct UtxoChangesResponse {
...
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct CongestionResponse {
+    /// The slot index for which the congestion estimate is provided.
+    pub slot_index: SlotIndex,
+    /// Indicates if a node is ready to issue a block in a current congestion or should wait.
+    pub ready: bool,
+    /// The cost in mana for issuing a block in a current congestion estimated based on RMC and slot index.
+    pub reference_mana_cost: String,
```
Comment (DaughterOfMars) [inline sdk/src/types/api/core/response.rs]: amount?

Comment (thibault-martinez) [reply sdk/src/types/api/core/response.rs]: Changed

File: sdk/src/types/api/core/response.rs [hunk 2]
```diff
@@ -358,7 +358,9 @@ pub struct CongestionResponse {
     /// Indicates if a node is ready to issue a block in a current congestion or should wait.
     pub ready: bool,
     /// The cost in mana for issuing a block in a current congestion estimated based on RMC and slot index.
-    pub reference_mana_cost: String,
+    #[cfg_attr(feature = "serde", serde(with = "crate::utils::serde::string"))]
```
Comment (DaughterOfMars) [inline sdk/src/types/api/core/response.rs]: the cfg attr isn't actually necessary

File: sdk/src/client/node_api/core/routes.rs [hunk 3]
```diff
@@ -91,6 +91,18 @@ impl ClientInner {
             .await
     }
 
+    /// Checks if the account is ready to issue a block.
```
Comment (kwek20) [inline sdk/src/client/node_api/core/routes.rs:94]: TIP writes Check, not Checks 🏃‍♂️

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:94]: It's fine, we don't alway keep 100% like the TIPs

File: sdk/src/types/api/core/response.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/types/api/core/response.rs]: How so ?

Comment (DaughterOfMars) [reply sdk/src/types/api/core/response.rs]: its just ignored if serde isn't derived

Comment (thibault-martinez) [reply sdk/src/types/api/core/response.rs]: Changed

**Review by DaughterOfMars — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Pull request merged**